modules/light-clients/08-wasm/testing/simapp/customquery/handler.go

```diff
@@ -0,0 +1,67 @@
+package customquery
+
+/*
+ * This custom query handler is used to aggregate public keys and verify a signature using BLS.
+ * It is used by the 08-wasm union light client, which we we use in the solidity IBC Eureka e2e tests.
+ * The code here is taken from here: https://github.com/unionlabs/union/tree/main/uniond/app/custom_query
+ */
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	MessageSize = 32
+)
+
+type CustomQuery struct {
+	AggregateVerify *QueryAggregateVerify `json:"aggregate_verify,omitempty"`
+	Aggregate       *QueryAggregate       `json:"aggregate,omitempty"`
+}
+type QueryAggregate struct {
+	PublicKeys [][]byte `json:"public_keys"`
+}
+type QueryAggregateVerify struct {
+	PublicKeys [][]byte `json:"public_keys"`
+	Signature  []byte   `json:"signature"`
+	Message    []byte   `json:"message"`
+}
+
+func CustomQuerier() func(sdk.Context, json.RawMessage) ([]byte, error) {
+	return func(ctx sdk.Context, request json.RawMessage) ([]byte, error) {
+		var customQuery CustomQuery
+		err := json.Unmarshal([]byte(request), &customQuery)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse custom query %v", err)
+		}
+
+		switch {
+		case customQuery.Aggregate != nil:
+			aggregatedPublicKeys, err := AggregatePublicKeys(customQuery.Aggregate.PublicKeys)
+			if err != nil {
+				return nil, fmt.Errorf("failed to aggregate public keys %v", err)
+			}
+
+			return json.Marshal(aggregatedPublicKeys.Marshal())
+		case customQuery.AggregateVerify != nil:
+			if len(customQuery.AggregateVerify.Message) != MessageSize {
+				return nil, fmt.Errorf("invalid message length, must be a 32bytes hash: %x", customQuery.AggregateVerify.Message)
```

Comment:
[nit] I guess it doesn't hurt to add the obtained len to the error message. Also, you could use MessageSize instead of an hardcoded "32" in the message

Reply:
Yeah, added it now :)